src/actuator/ActuatorSimple.C

```diff
@@ -0,0 +1,1204 @@
+
+  }
+
+  // Reset thrust and torque
+  for (size_t iBlade = 0; iBlade < n_simpleblades_; ++iBlade) {
+    BladeTotalLift[iBlade] = 0.0;
+    BladeTotalDrag[iBlade] = 0.0;
+    BladeAvgAlpha[iBlade]  = 0.0;
+    for (int j = 0; j < nDim; j++) {
+      torque[iBlade][j] = 0.0;
+      thrust[iBlade][j] = 0.0;
+      BladeAvgWS2D[iBlade][j] = 0.0;
+    }
+  }
+
+  // if disk average azimuthally
+  // apply the forcing terms
+  execute_class_specific(nDim, coordinates, actuator_source, dualNodalVolume);
+
+  // Write the thrust outputs
+  NaluEnv::self().naluOutputP0() << " -- START BLADE summary --" <<std::endl;
+  // Outputs from source terms
+  for (size_t iBlade =0; iBlade < n_simpleblades_; iBlade++) {
+    auto bladeInfo =
+      dynamic_cast<ActuatorSimpleInfo*>(actuatorInfo_.at(iBlade).get());
+
+    if (NaluEnv::self().parallel_rank() == bladeInfo->runOnProc_) 
+      NaluEnv::self().naluOutput()
+	<< " ALS Blade "<<iBlade<<" Force: "<<std::setprecision(8)
+	<< thrust[iBlade][0] << " "
+	<< thrust[iBlade][1] << " " << thrust[iBlade][2] << " " << std::endl;
+  }
+  // Outputs from BEM theory
+  for (size_t iBlade =0; iBlade < n_simpleblades_; iBlade++) {
+    auto bladeInfo =
+      dynamic_cast<ActuatorSimpleInfo*>(actuatorInfo_.at(iBlade).get());
+    
+    if (NaluEnv::self().parallel_rank() == bladeInfo->runOnProc_) 
+      {
+	NaluEnv::self().naluOutput()
+	  << " BEM Blade "<<iBlade<<" Lift: "<<std::setprecision(8)
+	  << BladeTotalLift[iBlade]<<" Drag: "
+	  << BladeTotalDrag[iBlade]<<std::endl;
+      }
+  }
+  // AVG outputs
+  for (size_t iBlade =0; iBlade < n_simpleblades_; iBlade++) {
+    auto bladeInfo =
+      dynamic_cast<ActuatorSimpleInfo*>(actuatorInfo_.at(iBlade).get());
+    const size_t Npts = bladeInfo->num_force_pts_blade_;
+    if (NaluEnv::self().parallel_rank() == bladeInfo->runOnProc_) 
+      {
+	NaluEnv::self().naluOutput()
+	  << " AVG Blade "<<iBlade<<" Alpha: "<<std::setprecision(8)
+	  << BladeAvgAlpha[iBlade]/(float)Npts << " WS: "
+	  << BladeAvgWS2D[iBlade][0]/(float)Npts << " "
+	  << BladeAvgWS2D[iBlade][1]/(float)Npts << " "
+	  << BladeAvgWS2D[iBlade][2]/(float)Npts << " "
+	  << std::endl;
+      }
+  }
+  NaluEnv::self().naluOutputP0() << " -- END summary --" <<std::endl;
+
+
+  //throw std::runtime_error("ActuatorSimple: done execute_class_specific()");  //LCCSTOP
+
+  // parallel assemble (contributions from ghosted and locally owned)
+  const std::vector<const stk::mesh::FieldBase*> sumFieldVec(
+    1, actuator_source);
+  stk::mesh::parallel_sum_including_ghosts(bulkData, sumFieldVec);
+
+  const std::vector<const stk::mesh::FieldBase*> sumFieldG(1, g);
+  stk::mesh::parallel_sum_including_ghosts(bulkData, sumFieldG);
+}
+
+
+// Copmute the filtered lifting line theory correction
+void ActuatorSimple::filtered_lifting_line()
+{
+
+}
+
+// Creates bounding boxes around the subdomain of each processor
+void
+ActuatorSimple::populate_candidate_procs()
```

Comment:
we can probably lift this out as a free function taking a selector or part vector and returning the bounding box/processor vector.  Not necessary for this PR though

Reply:
I agree, we can do that as a separate PR.